### PR TITLE
simplifies some verbose lines of code.

### PR DESCRIPTION
**Pull request by lemire**
Patterns such as

```
auto out = std::string();
```

Are probably best avoided and replaced with...

```
std::string out;
```

This PR makes a few such changes.

### CODE REVIEW

File: benchmarks/benchmark_header.h [hunk 1]
```diff
@@ -32,11 +32,11 @@ bool file_exists(const char* filename) {
 }
 
 std::string read_file(std::string filename) {
-  constexpr auto read_size = std::size_t(4096);
+  constexpr std::size_t read_size = 4096;
```
Comment (anonrig) [inline benchmarks/benchmark_header.h:35]: Is std namespace really required?

**Review by anonrig — APPROVED** [review]

File: benchmarks/benchmark_header.h (hunk 1 above)
Comment (lemire) [reply benchmarks/benchmark_header.h:35]: No, it is not.

Comment (lemire) [reply benchmarks/benchmark_header.h]: Fixed in following PR.

Comment (lemire) [reply benchmarks/benchmark_header.h]: Scrap that, fixed in following commit.

**Review by anonrig — APPROVED** [review]

**Pull request merged**